### PR TITLE
Bring in loopback4-example-log-extension

### PR DESCRIPTION
**Pull request by bajtos**
The first commit 76558a6 brings in an exact copy of https://github.com/strongloop/loopback4-example-log-extension.

The second commit f4dd44e contains changes necessary to integrate the example with the rest of the monorepo.

See #836

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: packages/example-getting-started/package.json [hunk 1]
```diff
@@ -14,12 +14,12 @@
     "build:dist": "lb-tsc es2017",
     "build:dist6": "lb-tsc es2015",
     "build:apidocs": "lb-apidocs",
-    "clean": "lb-clean loopback-getting-started*.tgz dist dist6 package api-docs",
+    "clean": "lb-clean *example-getting-started*.tgz dist dist6 package api-docs",
```
Comment (virkt25) [inline packages/example-getting-started/package.json]: Is the `*` before the word `example` necessary?

File: packages/example-log-extension/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2017 StrongLoop and IBM API Connect
```
Comment (virkt25) [inline packages/example-log-extension/LICENSE]: 2018. I believe all the other packages use a slightly different header for LICENSE. I think this should use the same. 

```
Copyright (c) IBM Corp. 2017,2018. All Rights Reserved.
Node module: @loopback/example-log-extension
This project is licensed under the MIT License, full text below.

--------

MIT license
```

File: packages/example-log-extension/README.md [hunk 3]
```diff
@@ -0,0 +1,364 @@
+# loopback4-example-log-extension
```
Comment (virkt25) [inline packages/example-log-extension/README.md]: Should we change this to `example-log-extension`?

File: packages/example-log-extension/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
```
Comment (virkt25) [inline packages/example-log-extension/index.d.ts]: 2017, 2018.

(All files -- since others in loopback-next were already updated).

File: packages/example-log-extension/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+// Node module: loopback4-example-log-extension
```
Comment (virkt25) [inline packages/example-log-extension/index.d.ts]: `Node module: @loopback/example-log-extension`.

File: packages/example-log-extension/package.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "loopback4-example-log-extension",
```
Comment (virkt25) [inline packages/example-log-extension/package.json]: `@loopback/example-log-extension`.

File: packages/example-log-extension/package.json [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  "version": "1.0.0",
```
Comment (virkt25) [inline packages/example-log-extension/package.json]: example getting started is version as `4.0.0-alpha.0`. Should we use the same here?

File: packages/example-log-extension/package.json [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    "tslint:fix": "npm run tslint -- --fix",
+    "prepare": "npm run build",
+    "pretest": "npm run clean && npm run build:current",
+    "test": "lb-dist mocha --opts node_modules/@loopback/build/mocha.ts.opts 'DIST/test/unit/**/*.js' 'DIST/test/acceptance/**/*.js'",
+    "posttest": "npm run lint",
+    "verify": "npm pack && tar xf *example-log-extension*.tgz && tree package && npm run clean"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/strongloop/loopback4-example-log-extension.git"
```
Comment (virkt25) [inline packages/example-log-extension/package.json]: Point to loopback-next repo.

**Comment by shimks** [comment]
I've noticed that this repo doesn't use `MetadataInspector` from `@loopback/metadata`. Should we migrate this first and then look to refactor the repo?

File: packages/example-log-extension/test/mocha.opts [hunk 9]
```diff
@@ -0,0 +1,2 @@
+--recursive
```
Comment (virkt25) [inline packages/example-log-extension/test/mocha.opts]: This can be dropped since `package.json` uses `mocha.opts` from `@loopback/build`.

File: packages/example-log-extension/test/smoke.test.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback4-example-log-extension
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
```
Comment (virkt25) [inline packages/example-log-extension/test/smoke.test.ts]: I think it's safe to drop this file as well.

File: packages/example-getting-started/package.json [hunk 11]
```diff
@@ -14,12 +14,12 @@
     "build:dist": "lb-tsc es2017",
     "build:dist6": "lb-tsc es2015",
     "build:apidocs": "lb-apidocs",
```
Comment (shimks) [inline packages/example-getting-started/package.json]: Should we be creating apidocs for this repo?

File: packages/example-log-extension/LICENSE (hunk 2 above)
Comment (shimks) [inline packages/example-log-extension/LICENSE]: Should we also update the copyright years with this PR?

**Comment by virkt25** [comment]
@shimks I agree. It's probably best to create a issue to follow up on the refactor but this PR's scope should be migration only.

File: packages/example-getting-started/package.json (hunk 1 above)
Comment (bajtos) [reply packages/example-getting-started/package.json]: Yes. Here is the full tarball name as produced by `npm pack`:

```
loopback-example-getting-started-1.0.1-alpha.0.tgz
```

File: packages/example-log-extension/LICENSE (hunk 2 above)
Comment (bajtos) [reply packages/example-log-extension/LICENSE]: Good catch, I'll use `LICENSE` from getting-started example.

File: packages/example-log-extension/README.md (hunk 3 above)
Comment (bajtos) [reply packages/example-log-extension/README.md]: I'll use `@loopback/example-log-extension` for consistency with getting-started example (I'll fix package.json too).

File: packages/example-log-extension/index.d.ts (hunk 4 above)
Comment (bajtos) [reply packages/example-log-extension/index.d.ts]: Let's use `slt license` because that's the tool we will use in the future.

```
SLT_OWNER="IBM Corp." slt copyright
```

Unfortunately this tool sets the copyright year to 2018 because it uses `git` commit history to find out the date of the first commit. Do you think this is a problem?

File: packages/example-log-extension/package.json (hunk 7 above)
Comment (bajtos) [reply packages/example-log-extension/package.json]: Sure.

**Comment by bajtos** [comment]
@virkt25 @shimks I think I have addressed all your review comments in e80ddc1, LGTY now?

File: packages/rest/index.js [hunk 12]
```diff
@@ -4,4 +4,6 @@
 // License text available at https://opensource.org/licenses/MIT
 
 const nodeMajorVersion = +process.versions.node.split('.')[0];
-module.exports = nodeMajorVersion >= 7 ? require('./dist') : require('./dist6');
+module.exports = nodeMajorVersion >= 7 ?
```
Comment (virkt25) [inline packages/rest/index.js]: Please revert the changes to this file.
